aicsimageio/readers/ome_zarr_reader.py

```diff
@@ -0,0 +1,182 @@
+    OmeZarr images.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    fs_kwargs: Dict[str, Any]
+        Any specific keyword arguments to pass down to the fsspec created filesystem.
+        Default: {}
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            _ = fs  # TODO: How to use the FS to test for suppored image
+            ZarrReader(parse_url(path, mode="r"))
+            return True
+
+        except ValueError:
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        channel_names: Optional[Union[List[str], List[List[str]]]] = None,
+        fs_kwargs: Dict[str, Any] = {},
+    ):
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(
+            image,
+            enforce_exists=True,
+            fs_kwargs=fs_kwargs,
+        )
+
+        if not isinstance(self._fs, LocalFileSystem):
+            raise ValueError(
+                f"Cannot read OmeZarr from non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+        self._channel_names = channel_names
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            OmeZarr = ZarrReader(parse_url(self._path, mode="r")).zarr
+            self._scenes = tuple(
+                metadata_utils.generate_ome_image_id(i)
+                for i in range(len(OmeZarr.root_attrs["multiscales"]))
+            )
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+        return self._xarr_format(delayed=True)
+
+    def _read_immediate(self) -> xr.DataArray:
+        return self._xarr_format(delayed=False)
+
+    def _xarr_format(self, delayed: bool) -> xr.DataArray:
+
+        OmeZarr = ZarrReader(parse_url(self._path, mode="r")).zarr
+        image_data = OmeZarr.load(str(self.current_scene_index))
+        axes = OmeZarr.root_attrs["multiscales"][self.current_scene_index]["axes"]
+        dims = [sub["name"].upper() for sub in axes]
+
+        if not delayed:
+            image_data = image_data.compute()
+
+        channels = self._get_channel_names_for_scene(image_data.shape, dims)
+
+        coords = self._get_coords(
+            dims,
+            image_data.shape,
+            scene_index=self.current_scene_index,
+            channel_names=channels,
+        )
+
+        return xr.DataArray(
+            image_data,
+            dims=dims,
+            coords=coords,
+            attrs={
+                # TODO: ADD info from "multiscales" "omero"?
```

Comment:
Would this be useful / sufficient metadata to provide under the `UNPROCESSED` metadata key?
https://github.com/ome/ome-zarr-py/blob/685eb89ab009560f5e41bd7cd0b7c067d91e0a2e/ome_zarr/io.py#L106

Reply:
I added root_attrs to UNPROCESSED in commit [bd1b617](https://github.com/AllenCellModeling/aicsimageio/pull/493/commits/bd1b61741488f3957a8cee5cf52383e79babfaeb)